docs/pages/documentation/11-examples/example-twitter-streaming-public-tweets.md

```diff
@@ -0,0 +1,580 @@
+    });
+  }
+
+```
+
+## Crafting Simple Queries Using GraQL
+
+We will perform a query which will count the number of tweets a user has posted since the program started. It can be achieved it by utilizing the aggregate query feature. Graql has been chosen over the graph API for this task because it is declarative and therefore much easier to use for complex queries than the graph API.
+
+Let's look at how we can build it step-by-step, start by creating a `QueryBuilder` object which we will use to craft the query in GraQL.
+
+```java
+QueryBuilder qb = graknGraph.graql();
+```
+
+Now let's begin crafting the query. For this tutorial, let's create a `match` query where we retrieve both the `user` and `tweet`.
+
+We will bind them into `var`s which will be named... `user` and `tweet`, respectively. Notice how we deliberately assign the `var`s  identical names as the respective entity types. This is not a necessity and in practice, you are free to name them anything you want.
+
+Also, pay attention to how we also supply the `user-tweet-relation` relation as part of the condition.
+
+```java
+qb.match(
+  var("user").isa("user"),
+  var("tweet").isa("tweet"),
+  var().rel("posts", "user").rel("posted_by", "tweet").isa("user-tweet-relation"))
+```
+
+The query we've just defined will return every user and tweet along with their relations. We will use it as the basis of the aggregate query. 
+
+Let's do some aggregation over the result here. We will supply `"user"` and `count()` as the argument for `group()`, which essentially tells Grakn to group the result by username, and count the number of occurences per username.
+
+```java
+qb.match(
+  var("user").isa("user"),
+  var("tweet").isa("tweet"),
+  var().rel("posts", "user").rel("posted_by", "tweet").isa("user-tweet-relation")
+).aggregate(group("user", count()));
+```
+
+ The query will now return "the number of tweet a user has posted", which is what we want, as an object of type `Map<Concept, Long>`. To be able to conveniently iterate, we will transform it into the relatively more straightforward type `Stream<Map.Entry<String, Long>>`, i.e., a stream of pairs of username and tweet count.
+
+```java
+  // execute query
+  Map<Concept, Long> result = ((Map<Concept, Long>) q.execute());
+
+  // map Map<Concept, Long> into Stream<Map.Entry<String, Long>> before returning
+  ResourceType screenNameResourceType = graknGraph.getResourceType("screen_name");
+
+  Stream<Map.Entry<String, Long>> mapped = result.entrySet().stream().map(entry -> {
+    Concept key = entry.getKey();
+    Long value = entry.getValue();
+    String screenName = (String) key.asEntity().resources(screenNameResourceType).iterator().next().getValue();
+    return new HashMap.SimpleImmutableEntry<>(screenName, value);
+  });
+```
+
+Voila! Here's how `calculateTweetCountPerUser` should look like.
+
+```java
+public static Stream<Map.Entry<String, Long>> calculateTweetCountPerUser(GraknGraph graknGraph) {
```

Comment:
Yeah, I don't want to include any 3rd party library, and on the other hand creating a class just for this return type is too heavy-handed. Therefore this..

How about I add a remark that address this return type?